### PR TITLE
Implement the design spec

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What is the purpose of this pull request?
Implements the visual changes to match the specification provided by our designer.

#### What problem is this solving?
The component didn't have a design spec before.

#### How should this be manually tested?
[Workspace](https://category-menu--storecomponents.myvtex.com/_v/category-menu)

```sh
$ yarn test
```

#### Screenshots or example usage
screenshot-2018-4-27 https lucas--storecomponents myvtex com
screenshot-2018-4-27 https lucas--storecomponents myvtex com 1

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/CategorySubMenu.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import CategorySubMenuItems from './CategorySubMenuItems'
+import { categoryPropType } from '../propTypes'
+
+class CategorySubMenu extends Component {
```
Comment (ClaudivanFilho) [inline react/components/CategorySubMenu.js]: Doc Missing

File: react/components/CategorySubMenuItems.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import { categoryPropType } from '../propTypes'
+
+class CategorySubMenuItems extends Component {
```
Comment (ClaudivanFilho) [inline react/components/CategorySubMenuItems.js]: Doc missing

File: react/components/CategorySubMenu.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  render() {
+    return (
+      <div
+        className="vtex-category-sub-menu pa3 br2 br--bottom"
```
Comment (ClaudivanFilho) [inline react/components/CategorySubMenu.js]: Segue o padrão
`vtex-category-sub-menu` => `vtex-category-menu__sub-menu`

File: react/components/CategorySubMenuItems.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  render() {
+    const { name, href, categories } = this.props
+
+    return (
+      <div className="vtex-category-sub-menu__items-container">
```
Comment (ClaudivanFilho) [inline react/components/CategorySubMenuItems.js]: `vtex-category-sub-menu__items-container` -> `vtex-category-menu__sub-menu__items-container`

Comment (lucasecdb) [reply react/components/CategorySubMenuItems.js]: No [BEM](https://en.bem.info/methodology/quick-start/#nesting-1) é considerado má prática colocar um `element` dentro de outro

File: react/components/CategorySubMenu.js (hunk 3 above)
Comment (lucasecdb) [reply react/components/CategorySubMenu.js]: O sub-menu nesse caso já é um bloco, e não um elemento dentro do category-menu

File: react/components/CategoryItem.js [hunk 5]
```diff
@@ -12,54 +13,21 @@ class CategoryItem extends Component {
...
-  category: PropTypes.shape({
-    id: PropTypes.number.isRequired,
-    name: PropTypes.string.isRequired,
-    href: PropTypes.string.isRequired,
-    hasChildren: PropTypes.bool.isRequired,
-  }).isRequired,
+  category: categoryPropType.isRequired,
 }
 
 export default CategoryItem
```
Comment (brunojdo) [inline react/components/CategoryItem.js]: ⛏  `export default class... `

File: react/components/CategoryItem.js [hunk 6]
```diff
@@ -12,54 +13,21 @@ class CategoryItem extends Component {
...
-            </ul>
-          </div>
-        )}
+        {hasChildren && <CategorySubMenu subCategories={category.children} />}
       </div>
     )
   }
 }
 
 CategoryItem.propTypes = {
```
Comment (brunojdo) [inline react/components/CategoryItem.js]: put into class. `static`

File: react/components/CategorySubMenuItems.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+              </a>
+            </li>
+          ))}
+        </ul>
+      </div>
+    )
+  }
+}
+
+CategorySubMenuItems.propTypes = {
```
Comment (brunojdo) [inline react/components/CategorySubMenuItems.js]: static

File: react/components/CategorySubMenu.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+ * categories and it's own sub-categories
+ */
+export default class CategorySubMenu extends Component {
+  static propTypes = {
+    subCategories: PropTypes.arrayOf(categoryPropType).isRequired,
+  }
+
+  render() {
+    return (
+      <div
```
Comment (brunojdo) [inline react/components/CategorySubMenu.js]: I think we need a `NoSSR` here

Comment (lucasecdb) [reply react/components/CategorySubMenu.js]: why?

Comment (lucasecdb) [reply react/components/CategorySubMenu.js]: nevermind

**Pull request merged**